src/patch.js

```diff
@@ -21,48 +23,91 @@ export function patch(node, container, runFx) {
       } else if (
         value != null &&
         value !== false &&
-        lifecycleEvents.indexOf(name) === -1
+        lifecycleEventNames.indexOf(name) === -1
       ) {
         element.setAttribute(name, value);
       }
+
+      if (value == null || value === false) {
+        element.removeAttribute(name);
+      }
     }
   }
 
-  function createElement(node, parent) {
-    var nextElement;
-    if (Array.isArray(node)) {
-      var type = node[0];
-      var props = node[1] || {};
-      if (typeof type === "string") {
-        nextElement = document.createElement(type);
-        for (var i = 1; i < node.length; i++) {
-          var childElement = createElement(node[i], nextElement);
-          if (childElement) {
-            nextElement.appendChild(childElement);
+  function createElement(node, propsNode) {
+    var element =
+      typeof node === "string" || typeof node === "number"
+        ? document.createTextNode(node)
+        : document.createElement(node[0]);
+    element.vnode = node;
+    stack.push(function() {
+      // oncreate
+      runFx((propsNode[1] || {})[lifecycleEventNames[0]], element);
```

Comment:
@okwolf Interesting. Are lifecycle events managed effects? How would this work?

Reply:
Yes. `@hyperapp/fx` supports this already using `event` fx. This implementation is based on the same premise, but built-in.

Reply:
https://github.com/hyperapp/fx#event

Reply:
@okwolf I need to review that documentation to see why is this useful.

Reply:
The only limitation of `event` is that it only handles unary event handler functions, which for Hyperapp lifecycle events means only access to the `element`.